### PR TITLE
Move router and history out of docs into main

### PR DESCRIPTION
**Pull request by zakpatterson**
Add the react-router and html5 history api to the web project

Also add macro settings to web project, for this to work

### CODE REVIEW

**Comment by shadaj** [comment]
Thanks for the PR @zakpatterson! The `web` module is supposed to be just the bindings to ReactDOM and the tags API, so could you move the router facades into another subproject with `name := "slinky-react-router"`? Then the published name would line up with other facades like slinky-styled-components.

**Comment by zakpatterson** [comment]
Ok, think that's good. Let me know what you think. Hopefully this helps people a little; I was just copying the router/history code from the docs into my project.

File: reactrouter/build.sbt [hunk 1]
```diff
@@ -0,0 +1,7 @@
+enablePlugins(ScalaJSPlugin)
+
+name := "slinky-react-router"
+
+libraryDependencies += "org.scala-lang" % "scala-reflect" % scalaVersion.value
```
Comment (shadaj) [inline reactrouter/build.sbt]: What is this dependency needed for? It should be introduced transitively by the dependency on `core`.

File: README.md [hunk 2]
```diff
@@ -37,5 +39,5 @@ Slinky is split up into several submodules:
 + `coreIntellijSupport` contains IntelliJ-specific support for the `@react` macro annotation
 + `tests` contains the unit tests for the above modules (except native and vr which have local tests)
 + `docs` and `docsMacros` contains the documentation site, which is a Slinky app itself
-
++
```
Comment (shadaj) [inline README.md]: Nit: extra `+`

File: reactrouter/build.sbt (hunk 1 above)
Comment (zakpatterson) [reply reactrouter/build.sbt]: :heavy_check_mark:

File: README.md (hunk 2 above)
Comment (zakpatterson) [reply README.md]: Oops! :heavy_check_mark:

**Pull request merged**